core-data/selectors.js

```diff
@@ -0,0 +1,5 @@
+import { values } from 'lodash';
+
+export function getCategories( state ) {
+	return values( state );
```

Comment:
This will return a new value on each return, thereby breaking any component render purity.

We can define the shape of the reducer by the optimized access patterns:

- Array of categories? Probably not as efficient for lookup by ID, slug
- But as implemented assumes lookup by ID will be common. Is that accurate?

Reply:
> But as implemented assumes lookup by ID will be common. Is that accurate?

Hard to answer this question, probably not so common in Gutenberg, but could be in other contexts. I'm happy to refactor this though.

Reply:
@aduth any plans to develop `lodash-memize`? That would be a quite useful library with the code we deal recently 😃